src/WinRT.Runtime/Projections/IList.net5.cs

```diff
@@ -33,223 +34,485 @@ interface IVector<T> : IIterable<T>
     }
 }
 
+namespace System.Collections.Generic
+{
+    internal sealed class IListImpl<T> : global::System.Collections.Generic.IList<T>
+    {
+        private readonly IObjectReference _inner;
+
+        private volatile IObjectReference __iListObjRef;
+        private IObjectReference Make_IListObjRef()
+        {
+            global::System.Threading.Interlocked.CompareExchange(ref __iListObjRef, _inner.As<ABI.System.Collections.Generic.IList<T>.Vftbl>(), null);
+            return __iListObjRef;
+        }
+        private IObjectReference iListObjRef => __iListObjRef ?? Make_IListObjRef();
+
+        private volatile IObjectReference __iEnumerableObjRef;
+        private IObjectReference Make_IEnumerableObjRef()
+        {
+            global::System.Threading.Interlocked.CompareExchange(ref __iEnumerableObjRef, _inner.As<ABI.System.Collections.Generic.IEnumerable<T>.Vftbl>(), null);
+            return __iEnumerableObjRef;
+        }
+        private IObjectReference iEnumerableObjRef => __iEnumerableObjRef ?? Make_IEnumerableObjRef();
+
+        internal IListImpl(IObjectReference _inner)
+        {
+            this._inner = _inner;
+        }
+
+        public T this[int index] 
+        {
+            get => ABI.System.Collections.Generic.IListMethods<T>.Indexer_Get(iListObjRef, index);
+            set => ABI.System.Collections.Generic.IListMethods<T>.Indexer_Set(iListObjRef, index, value); 
+        }
+
+        public int Count => ABI.System.Collections.Generic.IListMethods<T>.get_Count(iListObjRef);
+
+        public bool IsReadOnly => ABI.System.Collections.Generic.IListMethods<T>.get_IsReadOnly(iListObjRef);
+
+        public IObjectReference NativeObject => iListObjRef;
```

Comment:
Are these 3 IWinRTObject properties used?  Class doesn't seem to implement IWinRTObject.